test/helpers/generators.go

```diff
@@ -53,11 +58,11 @@ func GenerateGateway(gatewayNSN types.NamespacedName, gatewayClass *gatewayv1.Ga
 }
 
 // GenerateGatewayConfiguration generates a GatewayConfiguration to be used in tests
-func GenerateGatewayConfiguration(gatewayConfigurationNSN types.NamespacedName) *operatorv1beta1.GatewayConfiguration {
+func GenerateGatewayConfiguration(namespace string) *operatorv1beta1.GatewayConfiguration {
 	return &operatorv1beta1.GatewayConfiguration{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace: gatewayConfigurationNSN.Namespace,
-			Name:      gatewayConfigurationNSN.Name,
+			Namespace: namespace,
+			Name:      uuid.NewString(),
```

Comment:
Nit: we might consider using `GenerateName` to make names have a bit more meaning (and also to be shorter).